pcdscalc/be_lens_calcs.py

```diff
@@ -937,3 +947,373 @@ def find_z_pos(energy, lens_set, spot_size_fwhm, material=None,
+            distance=z_offset - max(z_range),
+            fwhm_unfocused=beam_size_unfocused,
+            printsummary=False,
+        ) for set in sets])
+
+    sizes = sizes[good_sets]
+    effrads = effrads[good_sets]
+    distances = distances[good_sets]
+    foclens = foc_lens[good_sets]
+    transms = np.asarray(
+        [lens_transmission(r=ter, fwhm=beam_size_unfocused, energy=energy)
+            for ter in effrads])
+    Nlenses_s = np.sum(sets, 1)
+
+    # used for testing
+    num = []
+    f_m = []
+    min_um = []
+    max_um = []
+    t_percent = []
+
+    resstring = " N   f/m   Min/um   Max/um   T/%  Set  \n"
+    zips = list(
+        zip(sets, size_range_min, size_range_max,
+            effrads, transms, Nlenses_s, foclens)
+    )
+
+    t = "{:2d} {:5.2f} {:8.1f} {:8.1f} {:5.1f}  "
+    for n, z in enumerate(zips):
+        the_set, sizemin, sizemax, eff_rad, transm, n_lenses, foclen = z
+        logger.debug('eff_rad: %s', eff_rad)
+        logger.debug('n_lenses: %s', n_lenses)
+
+        num.append(n)
+        f_m.append(foclen)
+        min_um.append(sizemin * 1e6)
+        max_um.append(sizemax * 1e6)
+        t_percent.append(transm * 100)
+
+        resstring += t.format(n, foclen, sizemin * 1e6,
+                              sizemax * 1e6, transm * 100)
+        resstring += ", ".join(
+            [
+                "%d x %dum" % (setLensno, t_lens_radii[m] * 1e6)
+                for m, setLensno in enumerate(the_set)
+                if setLensno > 0
+            ]
+        )
+        resstring += "\n"
+    logger.info('\n %s', resstring)
+    return num, f_m, min_um, max_um, t_percent
+
+
+def lens_transmission(r, fwhm, n=1, energy=None, id_material="IF1",
+                      lens_thicknes=None):
+    """
+    Find the CRL (Compound Refractive Lens) transmission.
+
+    Parameters
+    ----------
+    radius : float
+        Effective radius of curvature
+    fwhm : float
+        Incident beam size on the lens in meters
+    num : int
+        Number of lenses in the stack
+    energy : number
+        Photon energy
+    id_material : str
+        Lens material
+
+    Returns
+    -------
+    trans : float
+    """
+    lens_thicknes = lens_thicknes or APEX_DISTANCE
+    id_material = alias.get(id_material, id_material)
+    waist = 2 * gaussian_fwhm_to_sigma(fwhm)
+    x = np.linspace(-2 * fwhm, 2 * fwhm, 101)
+    y = x
+    intensity = np.zeros((len(x), len(y)))
+    thickness = np.zeros((len(x), len(y)))
+    for i in range(len(x)):
+        for j in range(len(y)):
+            intensity[i, j] = (
+                np.abs(np.exp(-2 * (x[i] ** 2 + y[j] ** 2) / waist ** 2))
+                * 2
+                / waist ** 2
+                / np.pi
+                * (x[2] - x[1]) ** 2
+            )
+            thickness[i, j] = (x[i] ** 2 + y[j] ** 2) / r + n * lens_thicknes
+    d = density.get(id_material)
+    att_length = get_att_len(energy, id_material, d)
```

Comment:
Need to revise this one, I'm using the `get_att_len` here (and tested it with `get_att_len` in the old code too), but after doing some research I realized that this is not the same as `attenuationLength` function from https://github.com/pcdshub/old-hutch-configs/blob/18c7769d66a647ab558ad7ebc34d019fc98fbf48/xpp/prod/utilitiesCalc.py#L362 
which brings with it a few more other functions.... that require `periodictable`...